browser/src/Editor/NeovimEditor/NeovimEditor.tsx

```diff
@@ -431,7 +431,17 @@ export class NeovimEditor extends Editor implements IEditor {
             }
         })
 
-        this._neovimInstance.on("tabline-update", (currentTabId: number, tabs: any[]) => {
+        this._neovimInstance.on("tabline-update", async (currentTabId: number, tabs: any[]) => {
+            const atomicCalls = tabs.map((tab: any) => {
+                return ["nvim_call_function", ["tabpagebuflist", [tab.id]]]
+            })
+
+            const response = await this._neovimInstance.request("nvim_call_atomic", [atomicCalls])
+
+            tabs.map((tab: any, index: number) => {
+                tab.buffersInTab = response[0][index]
```

Comment:
Need to check that `nvim_call_atomic` will always return in the order its called, which is what I'm expecting.

@bryphe, would you happen to know is this is the case?

Reply:
Sorry I missed this - that is my understanding though, that the order is preserved. We at least rely on that in a few other places too 😄

Reply:
I had a look over the documents + the PR that added `nvim_call_atomic` and I saw no mention it would return in any other order so I presume we are okay!